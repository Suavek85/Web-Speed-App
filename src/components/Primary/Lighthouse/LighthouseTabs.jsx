import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'
import LightHouseTabItem from './LightHouseTabItem'

import styles from './Lighthouse.scss'
import { 
  mobileWidth, 
  smallMobileWidth 
} from'../../../scss/mediaqueries'

export default function LighthouseTabs(props) {
  const { handleActiveTab, activeTab } = props

  let cx = classNames.bind(styles)
  let getTabClass = cx({
    blockTabs: true,
    blockTabsMobile: useMediaQuery(mobileWidth),
    blockTabsSmallMobile: useMediaQuery(smallMobileWidth),
  })

  return (
    <div onClickCapture={ handleActiveTab } className={ getTabClass }>
      <LightHouseTabItem activeTab={activeTab} />
    </div>
  )
}

LighthouseTabs.propTypes = {
  handleActiveTab: PropTypes.func,
  activeTab: PropTypes.string,
}