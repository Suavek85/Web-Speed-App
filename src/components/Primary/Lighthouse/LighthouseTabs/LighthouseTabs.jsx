import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'
import LighthouseTabItem from '../LighthouseTabItem/LighthouseTabItem'

import styles from './LighthouseTabs.scss'
import { 
  smallTabletWidth, 
  smallMobileWidth 
} from'../../../../scss/mediaqueries'

export default function LighthouseTabs(props) {

  const { handleActiveTab, activeTab } = props

  const cx = classNames.bind(styles)
  const getTabClass = cx({
    blockTabs: true,
    blockTabsSmallTablet: useMediaQuery(smallTabletWidth),
    blockTabsSmallMobile: useMediaQuery(smallMobileWidth),
  })

  return (
    <div onClickCapture={ handleActiveTab } className={ getTabClass }>
      <LighthouseTabItem activeTab={activeTab} />
    </div>
  )
}

LighthouseTabs.propTypes = {
  handleActiveTab: PropTypes.func,
  activeTab: PropTypes.string,
}