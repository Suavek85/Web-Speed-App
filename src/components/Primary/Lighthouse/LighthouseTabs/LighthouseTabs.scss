@import '../../../../scss/styles';

.lighthouse__tabs {
  display: flex;
  width: 100%;
  height: 44px;
  border-bottom: $ws-border-light;

  &--mobile {
    height: 54px;
  }

  &--small-mobile {
    flex-direction: column;
    height: 250px;
  }

  &--small-mobile > div {
    width: auto;

    &:not(:last-child) {
      border-right: 0;
      border-bottom: $ws-border-light;
    }
  }
}

:export {
  blockTabs: lighthouse__tabs;
  blockTabsMobile: lighthouse__tabs--mobile;
  blockTabsSmallMobile: lighthouse__tabs--small-mobile;
}
