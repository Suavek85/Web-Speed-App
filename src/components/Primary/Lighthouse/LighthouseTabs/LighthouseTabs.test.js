import React from 'react'
import { shallow } from 'enzyme'
import LighthouseTabs from './LighthouseTabs'

describe('LighthouseTabContent snapshot', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<LighthouseTabs />)
    expect(wrapper).toMatchSnapshot()
  })

})