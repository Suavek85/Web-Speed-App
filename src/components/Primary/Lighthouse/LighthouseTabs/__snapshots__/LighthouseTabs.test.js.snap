// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LighthouseTabContent snapshot should match the snapshot 1`] = `
<div
  className="blockTabs"
>
  <LighthouseTabItem />
</div>
`;
